app/models/foreman_kubevirt/kubevirt.rb

```diff
@@ -6,7 +6,7 @@ class Kubevirt < ComputeResource
     alias_attribute :token, :password
     alias_attribute :namespace, :user
     validates :hostname, :api_port, :namespace, :token, :presence => true
-    before_save :test_connection
+    after_validation :connect unless Rails.env.test?
```

Comment:
What guarantees the validation as part of submitting the form ? Shouldn't that be the 'before_save' ?

Reply:
before save will not run the function as a validation when submitting the form